src/messaging/tests/TestReliableMessageProtocol.cpp

```diff
@@ -179,20 +190,30 @@ void CheckResendMessage(nlTestSuite * inSuite, void * inContext)
         1, // CHIP_CONFIG_RMP_DEFAULT_ACTIVE_RETRY_INTERVAL
     });
 
-    gSendMessageCount = 0;
-
-    err = exchange->SendMessage(Echo::MsgType::EchoRequest, std::move(buffer));
+    gLoopback.mSendMessageCount = 0;
+    gLoopback.mDropNextMessage  = true;
+    err                         = exchange->SendMessage(Echo::MsgType::EchoRequest, std::move(buffer));
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, gLoopback.mDroppedMessageCount == 1);
+    NL_TEST_ASSERT(inSuite, rm->TestGetCountRetransTable() == 1);
 
+    gLoopback.mDropNextMessage = true;
```

Comment:
assert that mDropNextMessage == false here. It is not immediately obvious that this has been reset (and we rely on the mDroppedMessageCount check above)